drivers/dma/dma_stm32u5.c

```diff
@@ -0,0 +1,757 @@
+
+	DMA_InitStruct.SrcDataWidth = table_p_size[index];
+
+	index = find_lsb_set(config->dest_data_size) - 1;
+	DMA_InitStruct.DestDataWidth = table_m_size[index];
+
+	DMA_InitStruct.SrcBurstLength = 1;
+	DMA_InitStruct.DestBurstLength = 1;
+	DMA_InitStruct.TransferEventMode = LL_DMA_TCEM_BLK_TRANSFER;
+	DMA_InitStruct.SrcAllocatedPort = LL_DMA_SRC_ALLOCATED_PORT0;
+	DMA_InitStruct.DestAllocatedPort = LL_DMA_DEST_ALLOCATED_PORT0;
+	DMA_InitStruct.LinkAllocatedPort = LL_DMA_LINK_ALLOCATED_PORT1;
+	DMA_InitStruct.LinkStepMode = LL_DMA_LSM_FULL_EXECUTION;
+	DMA_InitStruct.TriggerMode = LL_DMA_TRIGM_BLK_TRANSFER;
+	DMA_InitStruct.TriggerPolarity = LL_DMA_TRIG_POLARITY_MASKED;
+	DMA_InitStruct.TriggerSelection = 0;
+
+	if (stream->source_periph) {
+		DMA_InitStruct.BlkDataLength = config->head_block->block_size /
+					config->source_data_size;
+	} else {
+		DMA_InitStruct.BlkDataLength = config->head_block->block_size /
+					config->dest_data_size;
+	}
+
+	/* The request ID is stored in the dma_slot */
+	DMA_InitStruct.Request = config->dma_slot;
+
+	LL_DMA_Init(dma, dma_stm32_id_to_stream(id), &DMA_InitStruct);
+
+	LL_DMA_EnableIT_TC(dma, dma_stm32_id_to_stream(id));
+
+	/* Enable Half-Transfer irq if circular mode is enabled */
+	if (config->head_block->source_reload_en) {
+		LL_DMA_EnableIT_HT(dma, dma_stm32_id_to_stream(id));
+	}
+
+	return ret;
+}
+
+static int dma_stm32_reload(const struct device *dev, uint32_t id,
+					  uint32_t src, uint32_t dst,
+					  size_t size)
+{
+	const struct dma_stm32_config *config = dev->config;
+	DMA_TypeDef *dma = (DMA_TypeDef *)(config->base);
+	struct dma_stm32_stream *stream;
+
+	/* Give channel from index 0 */
+	id = id - STM32_DMA_STREAM_OFFSET;
+
+	if (id >= config->max_streams) {
+		return -EINVAL;
+	}
+
+	stream = &config->streams[id];
+
+	if (dma_stm32_disable_stream(dma, id) != 0) {
+		return -EBUSY;
+	}
+
+	switch (stream->direction) {
+	case MEMORY_TO_PERIPHERAL:
+		stm32_dma_set_mem_periph_address(dma,
+						 dma_stm32_id_to_stream(id),
+						 src, dst);
+		break;
+	case MEMORY_TO_MEMORY:
+	case PERIPHERAL_TO_MEMORY:
+		stm32_dma_set_periph_mem_address(dma,
+						 dma_stm32_id_to_stream(id),
+						 src, dst);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	if (stream->source_periph) {
+		LL_DMA_SetBlkDataLength(dma, dma_stm32_id_to_stream(id),
+				     size / stream->src_size);
+	} else {
+		LL_DMA_SetBlkDataLength(dma, dma_stm32_id_to_stream(id),
+				     size / stream->dst_size);
+	}
+
+	stm32_dma_enable_stream(dma, id);
+
+	return 0;
+}
+
+static int dma_stm32_start(const struct device *dev, uint32_t id)
+{
+	const struct dma_stm32_config *config = dev->config;
+	DMA_TypeDef *dma = (DMA_TypeDef *)(config->base);
+
+	/* Give channel from index 0 */
+	id = id - STM32_DMA_STREAM_OFFSET;
+
+	/* Only M2P or M2M mode can be started manually. */
+	if (id >= config->max_streams) {
+		return -EINVAL;
+	}
+
+	dma_stm32_clear_stream_irq(dev, id);
+
+	stm32_dma_enable_stream(dma, id);
```

Comment:
In this driver what is the difference between these namespaces `dma_stm32` and `stm32_dma`